modules/core/src/smithy4s/internals/Hinted.scala

```diff
@@ -55,6 +55,20 @@ case class Hinted[F[_], A](hints: Hints, make: Hints => F[A]) {
       (h: Hints) => f(make(hints ++ h), other.make(other.hints ++ h))
     )
 
+  def emap[B](
+      f: A => Either[schema.ConstraintError, B]
+  )(implicit C: Covariant[F]): Hinted[F, B] =
+    Hinted(
+      hints,
+      h =>
+        C.map(make(h))(a =>
+          f(a) match {
+            case Left(constraintError) => throw constraintError
```

Comment:
is `throw` definitely safe here? If `F` doesn't suspend (e.g. Either) that might misbehave

Reply:
Dude, it's very much wip xD but yeah it's not the correct location to throw